os/logm/logm_process.c

```diff
@@ -27,20 +27,19 @@
 #include "logm_test.h"
 #endif
 
-volatile int logm_bufsize = LOGM_BUFFER_SIZE;
+uint8_t logm_status;
+int logm_bufsize = LOGM_BUFFER_SIZE;
+char * g_logm_rsvbuf = NULL;
 volatile int logm_print_interval = LOGM_PRINT_INTERVAL * 1000;
-char *volatile g_logm_rsvbuf = NULL;
-volatile uint8_t logm_status;
 
 static int logm_change_bufsize(void)
```

Comment:
Function style is very strange.
logm_change_bufsize(int bufsize) {
   return OK or Error instead of 0 or -1
}